picasso/src/test/java/com/squareup/picasso/BitmapHunterTest.java

```diff
@@ -873,6 +873,28 @@
     assertThat(shadowBitmap.getCreatedFromBitmap()).isNotSameAs(source);
   }
 
+  @Test public void onlyScaleDownResizeWidthIs0() {
+    Bitmap source = Bitmap.createBitmap(50, 50, ARGB_8888);
+    Request data = new Request.Builder(URI_1).resize(0, 60).onlyScaleDown().build();
+    Bitmap result = transformResult(data, source, 0);
+    assertThat(result).isSameAs(source);
+
+    ShadowBitmap shadowBitmap = shadowOf(result);
+    assertThat(shadowBitmap.getCreatedFromBitmap()).isNull();
+    assertThat(shadowBitmap.getCreatedFromBitmap()).isNotSameAs(source);
```

Comment:
This test logic is wrong (it looks copy/pasted). If we already check the value is null there's no reason to check it against something else.

I suspect it was meant to be `assertThat(result).isNotSameAs(source)` in the original test and here it should be `isSameAs`.